java/ql/src/semmle/code/java/dataflow/SSA.qll

```diff
@@ -385,11 +385,7 @@ private module SsaImpl {
       ma.getCaller() = c1 and
       m2 = viableImpl(ma) and
       not m2.isStatic() and
-      (
-        not exists(ma.getQualifier()) or
-        ma.getQualifier() instanceof ThisAccess or
-        ma.getQualifier() instanceof SuperAccess
-      ) and
+      ma.isOwnMethodAccess() and
       c1.getDeclaringType() = t1 and
       if t1 instanceof InnerClass
       then
```

Comment:
I am not sure whether the following `getQualifier()` checks could be simplified as well.

Reply:
It could potentially be eliminated as it ought to be superfluous with this change (the `then` branch attempts to capture exactly the difference between the simple syntactic checks removed by this change and the more involved `isOwnMethodAccess()`). However, this change in isolation does change behaviour (it shouldn't), which indicates a bug - quite possibly in `isOwnMethodAccess()`, and the SSA code is quite critical, so I don't think we should change this code without a proper understanding of those corner-cases and possibly a bug-fix to `isOwnMethodAccess()`. So let's revert this particular change for now.